builder/amazon/common/step_source_ami_info.go

```diff
@@ -103,7 +103,7 @@ func (s *StepSourceAMIInfo) Run(state multistep.StateBag) multistep.StepAction {
 
 	// Enhanced Networking (SriovNetSupport) can only be enabled on HVM AMIs.
 	// See http://goo.gl/icuXh5
-	if s.EnhancedNetworking && *image.VirtualizationType != "hvm" {
+	if (s.EnhancedNetworkingType == "sriov" || s.EnhancedNetworkingType == "ena") && *image.VirtualizationType != "hvm" {
```

Comment:
this is fine but I think `if s.EnhancedNetworkingType != "" && *image.VirtualizationType != "hvm"` would be simpler to understand and future-proof